src/catalog/src/builtin.rs

```diff
@@ -3278,6 +3278,49 @@ GROUP BY object_id, collection_timestamp",
     access: vec![PUBLIC_SELECT],
 });
 
+pub static MZ_RECENT_STORAGE_USAGE: Lazy<BuiltinView> = Lazy::new(|| BuiltinView {
+    name: "mz_recent_storage_usage",
+    schema: MZ_INTERNAL_SCHEMA,
+    oid: oid::VIEW_MZ_RECENT_STORAGE_USAGE_OID,
+    column_defs: Some("object_id, size_bytes"),
+    sql: "
+WITH
+
+recent_storage_usage_by_shard AS (
+    SELECT shard_id, size_bytes, collection_timestamp
+    FROM mz_internal.mz_storage_usage_by_shard
+    -- Restricting to the last 6 hours makes it feasible to index the view.
+    WHERE collection_timestamp + '6 hours' >= mz_now()
+),
+
+most_recent_collection_timestamp_by_shard AS (
+    SELECT shard_id, max(collection_timestamp) AS collection_timestamp
+    FROM recent_storage_usage_by_shard
+    GROUP BY shard_id
+)
+
+SELECT
+    object_id,
+    sum(size_bytes)::uint8
```

Comment:
Does this cast cause the SUM logic to use a uint8, or does it sum with the same type as size_bytes then cast the sum'd result to uint8?

Reply:
> does it sum with the same type as size_bytes then cast the sum'd result to uint8?

It sums with one size larger than `size_bytes` and then casts to `uint8`. This matches what we do in `mz_storage_usage`—the thinking was that there's no way this can ever overflow a `uint8` because that would be an astronomical amount of data.